src/app/common/services/confirm-dialog.service.ts

```diff
@@ -0,0 +1,20 @@
+import {Injectable} from '@angular/core';
+import {MatDialog, MatDialogRef} from '@angular/material/dialog';
+import {ConfirmationModalComponent} from '../components/confirmation-modal/confirmation-modal.component';
+import {IConfirmationModalData} from '../model/confirmation-modal-data';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConfirmDialogService {
+
+  public constructor(private readonly dialog: MatDialog) {
+  }
+
+  public openConfirmModal(message: IConfirmationModalData): MatDialogRef<ConfirmationModalComponent, boolean> {
```

Comment:
I guess `message` is not the best name here. Message would be some string. This is configuration for the confirmation modal component. So I would suggest name it like `config: IConfirmationModalConfig`